src/internal/pjsdb/queue_crud.go

```diff
@@ -42,10 +43,50 @@ func DequeueAndProcess(ctx context.Context, tx *pachsql.Tx, programHash []byte)
 		WHERE pjs.jobs.id = updated.id
 		RETURNING pjs.jobs.id
 	`, programHash).Scan(&jobID); err != nil {
+		// todo(muyang): should not return an error, just await
 		if errors.Is(err, sql.ErrNoRows) {
 			return 0, &DequeueFromEmptyQueueError{ID: string(programHash)}
 		}
 		return 0, errors.Wrap(err, "dequeue and process")
 	}
 	return jobID, nil
 }
+
+// CompleteError is called when job processing has an error. It updates job err code and
+// done timestamp in database.
+func CompleteError(ctx context.Context, tx *pachsql.Tx, jobID JobID, errCode pjs.JobErrorCode) error {
```

Comment:
I think this should be in job crud because you're marking a *job* as errored.

Based on that, this should be called ErrorJob or maybe FailJob.